### ISSUE TITLE
Shadow plugin incompatible with Gradle 7.0 - transformers's property missing an input or output annotation

### ISSUE THREAD

**Issue by avillev** [johnrengelman/shadow#665]
Please check the [User Guide](http://imperceptiblethoughts.com/shadow) before submitting "how do I do 'x'?" questions!

### Shadow Version

6.1.0

### Gradle Version

7.0

### Expected Behavior

To function

### Actual Behavior

This error happens:

`
Some problems were found with the configuration of task ':utils:utils-shaded-ahc:shadowJar' (type 'ShadowJar').
  - Type 'ShadowJar' property 'transformers.$0.keyTransformer' is missing an input or output annotation.

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.internal.execution.WorkValidationException: Some problems were found with the configuration of task ':utils:utils-shaded-ahc:shadowJar' (type 'ShadowJar').
  - Type 'ShadowJar' property 'transformers.$0.keyTransformer' is missing an input or output annotation.
    
    Reason: A property without annotation isn't considered during up-to-date checking.
    
    Possible solutions:
      1. Add an input or output annotation.
      2. Mark it as @Internal.
    
    Please refer to https://docs.gradle.org/7.0/userguide/validation_problems.html#missing_annotation for more details about this problem.
  - Type 'ShadowJar' property 'transformers.$0.mappings' is missing an input or output annotation.
    
    Reason: A property without annotation isn't considered during up-to-date checking.
    
    Possible solutions:
      1. Add an input or output annotation.
      2. Mark it as @Internal.
    
    Please refer to https://docs.gradle.org/7.0/userguide/validation_problems.html#missing_annotation for more details about this problem.
  - Type 'ShadowJar' property 'transformers.$0.mergeSeparator' is missing an input or output annotation.
    
    Reason: A property without annotation isn't considered during up-to-date checking.
    
    Possible solutions:
      1. Add an input or output annotation.
      2. Mark it as @Internal.
    
    Please refer to https://docs.gradle.org/7.0/userguide/validation_problems.html#missing_annotation for more details about this problem.
  - Type 'ShadowJar' property 'transformers.$0.mergeStrategy' is missing an input or output annotation.
    
    Reason: A property without annotation isn't considered during up-to-date checking.
    
    Possible solutions:
      1. Add an input or output annotation.
      2. Mark it as @Internal.
    
    Please refer to https://docs.gradle.org/7.0/userguide/validation_problems.html#missing_annotation for more details about this problem.
  - Type 'ShadowJar' property 'transformers.$0.paths' is missing an input or output annotation.
    
    Reason: A property without annotation isn't considered during up-to-date checking.
    
    Possible solutions:
      1. Add an input or output annotation.
      2. Mark it as @Internal.
    
    Please refer to https://docs.gradle.org/7.0/userguide/validation_problems.html#missing_annotation for more details about this problem.
`

### Gradle Build Script(s)

```groovy
import com.github.jengelman.gradle.plugins.shadow.transformers.PropertiesFileTransformer

final pomName = 'RAPID General asynchttpclient Shaded utility libraries'

//Shadow jar definition
shadowJar {
    manifest { inheritFrom project.tasks.manifestTask.manifest }
    archiveBaseName.set(artifactName)
    //No classifier, no -all on the jar (works only with the deprecated field because of https://github.com/johnrengelman/shadow/issues/463)
    classifier = ''

    //Relocation of the dependencies
    relocate 'org.asynchttpclient.',    'rapid.shaded.org.asynchttpclient.'
    relocate 'io.netty.',               'rapid.shaded.io.netty.'
    relocate 'com.typesafe.netty.',     'rapid.shaded.com.typesafe.netty.'
    relocate 'org.reactivestreams.',    'rapid.shaded.org.reactivestreams.'

    dependencies {
        exclude(dependency('io.netty::.*'))
        exclude(dependency('org.slf4j::.*'))
    }

    exclude 'javax/activation/**'
    exclude 'com/sun/activation/**'

    exclude 'META-INF/maven/com.sun.activation/**'

    transform(PropertiesFileTransformer) {
        keyTransformer = { key ->
            key.replaceAll('^(org\\.asynchttpclient\\..*)$', 'rapid.shaded.$1')
        }
    }
}

dependencies {
    implementation(Libraries.ahc)
    api(Libraries.slf4j)
}

ext.excluded_dependencies = [Libraries.ahc]
```

**Comment by VitaliyKulikov** [comment]
seems, something in progress https://github.com/johnrengelman/shadow/pull/657

**Comment by maxm123** [comment]
This was fixed in #647 and will be included in the next release.

**Comment by breskeby** [comment]
@maxm123 Any plans already when 6.2 will be released?
